pkg/dss/scd/store.go

```diff
@@ -2,14 +2,34 @@ package scd
 
 import (
 	"context"
+	"time"
 
 	"github.com/golang/geo/s2"
 	dssmodels "github.com/interuss/dss/pkg/dss/models"
 	scdmodels "github.com/interuss/dss/pkg/dss/scd/models"
 )
 
-// Store abstracts interactions with a backing data store.
-type Store interface {
+// OperationStore abstracts operation-specific interactions with the backing data store.
+type OperationStore interface {
+	// GetOperation returns the operation identified by "id".
+	GetOperation(ctx context.Context, id scdmodels.ID) (*scdmodels.Operation, error)
+
+	// DeleteOperation deletes the operation identified by "id" and owned by "owner".
+	// Returns the deleted Operation and all Subscriptions affected by the delete.
+	DeleteOperation(ctx context.Context, id scdmodels.ID, owner dssmodels.Owner) (*scdmodels.Operation, []*scdmodels.Subscription, error)
+
+	// UpsertOperation inserts or updates an operation using key as a fencing
+	// token. If operation does not reference an existing subscription, an
+	// implicit subscription with parameters notifySubscriptionForConstraints
+	// and subscriptionBaseURL is created.
+	UpsertOperation(ctx context.Context, operation *scdmodels.Operation, key []scdmodels.OVN, notifySubscriptionForConstraints bool, subscriptionBaseURL string) (*scdmodels.Operation, []*scdmodels.Subscription, error)
+
+	// SearchOperations returns all operations ownded by "owner" in "cells".
+	SearchOperations(ctx context.Context, cells s2.CellUnion, altitudeLower *float64, altitudeUpper *float64, earliest *time.Time, latest *time.Time) ([]*scdmodels.Operation, error)
```

Comment:
Could we add a dss.models.Geometry interface implementation that has a CellUnion and then have SearchOperations just accept a dss.models.Volume4D?

Reply:
Yup, I'll rework the interfaces to that extent.